NDTensors/src/lib/Sectors/src/categories.jl

```diff
@@ -0,0 +1,192 @@
+
+"""
+Representation of a category or group, storing its
+name and the dimension of its defining representation.
+One can also specify a quantum deformation "level".
+
+For example, Category("SU",2) is the group SU(2). 
+Category("Fib") is the Fibonacci category. 
+Category("SU",2,3) is the "quantum group" SU(2) level 3.
+"""
+struct Category
+  basename::String7
+  N::Int
+  level::Int
+end
+
+Category(basename, N::Int=0) = Category(basename, N, 0)
+
+Category() = Category("")
+
+basename(C::Category) = C.basename
+groupdim(C::Category) = C.N
+level(C::Category) = C.level
+
+isactive(C::Category) = length(C.basename) > 0
+
+function name(C::Category)
+  if groupdim(C) == 0
+    return C.basename
+  elseif level(C) == 0
+    return "$(C.basename)($(groupdim(C)))"
+  else
+    return "$(C.basename)($(groupdim(C)))_$(level(C))"
+  end
+  return error("Unexpected case")
+end
+
+Base.show(io::IO, C::Category) = print(io, name(C))
+
+nvals(::Any) = 1
+nvals(C::Category) = nvals(CategoryName(C))
+
+function fusion_rule(C::Category, a, b)
+  if nvals(C) == 1
+    return fusion_rule(CategoryName(C), a[1], b[1])
+  end
+  return fusion_rule(CategoryName(C), a, b)
+end
+
+#
+# Version of Category type but with
+# basename held statically as a 
+# type parameter. Mostly for internal use.
+#
+struct CategoryName{N}
+  N::Int
```

Comment:
Also, another design to consider is:
```julia
struct Category{Name}
  name::Name
  groupdim::Int
  level::Int
end
```
`Category("SU", 2)` would make the current `Category` object, and `Category(CategoryName"SU", 2)` would make the current `CategoryName` object. The new `CategoryName` would be a singleton type with no fields, like the `OpName` object in `ITensors.jl`.

My motivation for that change is that the current design has two slightly strange aspects that don't feel right to me:
1. `CategoryName` and `Category` are almost the same type except the name is static, but in the current design they are two different types.
2. `CategoryName` isn't a great name for the current type, since it isn't just the name but also has the group dimension and level.

Reply:
It seems like the current overloading syntax for `fusion_rule` could still be used, but instead:
```julia
fusion_rule(::Category"U", n1, n2) = [(n1 + n2)]
```
and `Category"U"` could be the type of a `Category` with a static name of type `CategoryName"U"`.

Reply:
We could have an interface for converting back and forth between `Category` with dynamic and static names, such as:
```julia
static(Category("SU", 2)) == Category(CategoryName("SU"), 2)
dynamic(Category(CategoryName("SU"), 2)) == Category("SU", 2)
```